server/shared/src/main/scala/org/http4s/server/middleware/BodyCache.scala

```diff
@@ -66,7 +66,7 @@ object BodyCache {
     apply(app)(identity, _ => identity)(FunctionK.id)
 
   private def compileBody[F[_]: Concurrent](req: Request[F]): F[Request[F]] =
-    req.toStrict
+    req.unsafeToStrict
```

Comment:
I wonder, shouldn't the `BodyCache` take into account body length and timeout also? Probably, it's worth creating a 'safe' version. If someone else agreed with that, I might work on it too. But that is out of the scope of this PR for sure.

Reply:
See some discussion about that here:
- https://github.com/http4s/http4s/pull/5321#issuecomment-935245046